src/database.ts

```diff
@@ -1307,6 +1359,61 @@ class Database extends GrpcServiceObject {
       callback!(err, transaction);
     });
   }
+
+  /**
+   * Query object for listing database operations.
+   *
+   * @typedef {object} ListDatabaseOperationsRequest
+   * @property {boolean} [autoPaginate=true] Have pagination handled
+   *     automatically.
+   * @property {number} [maxApiCalls] Maximum number of API calls to make.
+   * @property {number} [maxResults] Maximum number of items to return.
+   * @property {number} [pageSize] Maximum number of results per page.
+   * @property {string} [pageToken] A previously-returned page token
+   *     representing part of the larger set of results to view.
+   */
+  /**
+   * @typedef {array} ListDatabaseOperationsResponse
+   * @property {IOperation[]} 0 Array of {@link IOperation} instances.
+   * @property {object} 1 The full API response.
+   */
+  /**
+   * List pending and completed operations for the database.
+   *
+   * @see {@link Instance.listDatabaseOperations}
+   *
+   * @param query query object for listing database operations.
+   * @returns {Promise<ListDatabaseOperationsResponse>} when resolved, contains
+   *     a paged list of database operations.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = instance.database('my-database');
+   * const [operations] = await database.listDatabaseOperations();
+   */
+  async listDatabaseOperations(
+    query?: ListDatabaseOperationsRequest
+  ): Promise<ListDatabaseOperationsResponse> {
+    // Create a query that lists database operations only on this database from
+    // the instance. Operation name will be prefixed with the database path for
+    // all operations on this database
+    let dbSpecificFilter =
+      `(metadata.@type:CreateDatabaseMetadata AND metadata.database:${this.formattedName_}) OR ` +
```

Comment:
This will no longer be complete if additional database operation types were to be added in the future. It's probably better to filter using the name of the operation. The names of all database operations start with the name of the database, so it has the following form: `projects/p/instances/i/database/d/operations/op-id`

This means that you can list all operations of a database by listing all operations whose name start with the name of the database. See https://github.com/googleapis/java-spanner/blob/ed3874afcf55fe7381354e03dab3a3b97d7eb520/google-cloud-spanner/src/main/java/com/google/cloud/spanner/Database.java#L136 for an example.